### PR TITLE
FIO-4551 added unique IDs inside the Field Sets

### PR DESCRIPTION
**Pull request by HannaKurban**
## Link to Jira Ticket

https://formio.atlassian.net/browse/FIO-4551

## Description
Added unique IDs to the fields inside the NestedComponents in the Data Grid rows

### CODE REVIEW

File: src/components/fieldset/Fieldset.js [hunk 1]
```diff
@@ -40,4 +41,11 @@ export default class FieldsetComponent extends NestedComponent {
     super(...args);
     this.noField = true;
   }
+
+  createComponent(component, options, data, before) {
+    const comp = super.createComponent(component, options, data, before);
+    if (document.querySelectorAll(`[id^=${comp.id}]`).length) {
```
Comment (travist) [inline src/components/fieldset/Fieldset.js:47]: This seems really strange. Why didn't createComponent add the id in the first place? Why do we need to add a querySelectorAll here?

**Review by travist — CHANGES_REQUESTED** [review]

File: src/components/fieldset/Fieldset.js (hunk 1 above)
Comment (HannaKurban) [reply src/components/fieldset/Fieldset.js]: removed querySelectorAll

**Comment by HannaKurban** [comment]
all tests are passed

File: src/components/fieldset/Fieldset.js [hunk 2]
```diff
@@ -40,4 +41,9 @@ export default class FieldsetComponent extends NestedComponent {
     super(...args);
     this.noField = true;
   }
+
+  createComponent(component, options, data, before) {
+    component.id =  getRandomComponentId();
+    return super.createComponent(component, options, data, before);
```
Comment (TanyaGashtold) [inline src/components/fieldset/Fieldset.js:47]: I have checked the ticket. Does this issue effect only Fieldset component? I am pretty sure that all nested components (like Panel, Container and other) inside NestedArray components have the same issue with ids. It worth moving the fix to the NestedComponent/NestedDataComponent/NestedArrayComponent level.

**Review by TanyaGashtold — CHANGES_REQUESTED** [review]

File: src/components/fieldset/Fieldset.js (hunk 2 above)
Comment (HannaKurban) [reply src/components/fieldset/Fieldset.js]: added the fix tio the NestedComponent

**Review by TanyaGashtold — APPROVED** [review]

**Review by TanyaGashtold — APPROVED** [review]

**Pull request merged**